src/ZChain.Core/Tree/CpuMiner.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ZChain.Core.Tree
+{
+    public class CpuMiner : IMiner
```

Comment:
Consider a gradual refactoring...make this file reflect current mining setup, before more radical refactoring.